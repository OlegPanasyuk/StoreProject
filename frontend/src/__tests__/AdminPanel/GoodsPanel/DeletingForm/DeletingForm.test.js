import React from 'react';
import DeleteFormConnected, { DeleteForm } from '../../../../AdminPanel/GoodsPanel/DeleteForm';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    onHide: () => { },
    item: {
        id: 1
    }
}) {
    let component = shallow(<DeleteForm {...props} />);
    return { component, props };
}

describe('DeleteForm --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(DeleteForm.propTypes.item).toBe(PropTypes.object);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('DeleteForm ', () => {
    let initialState = {
        adminPanel_goodsPanel: {
            deleteItem: {
                show: false,
                id: null
            }
        }
    };

    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><DeleteFormConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(DeleteFormConnected).length).toEqual(1);
    });
});