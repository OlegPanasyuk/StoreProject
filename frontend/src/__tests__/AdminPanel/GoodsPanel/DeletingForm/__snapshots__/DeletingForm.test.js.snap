// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteForm --- Snapshot Component must be rendered 1`] = `
<Bootstrap(Modal)
  onHide={[Function]}
  show={true}
>
  <ModalHeader
    closeButton={false}
    closeLabel="Close"
  >
    <ModalTitle>
      Delete this object?
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    className="d-flex justify-content-end"
  >
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Delete
    </Button>
    <Button
      active={false}
      className="ml-3"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="light"
    >
      Cancel
    </Button>
  </ModalBody>
</Bootstrap(Modal)>
`;
